CDCgov/prime-reportstream
# Fix to issue that submission details is returning tons of repeated data

This PR fixes an issue that the Submission Details query was returning tons of extraneous repeats of reports.   See ticket #6454.

### How to Test

#### How to test localhost:

Generate a bunch of data going to several receivers, into a file called ./junk/mix.csv:

`./prime data --input-fake 30 --input-schema primedatainput/pdi-covid-19 --output ./junk/mix.csv --target-states MD,HI,FL
`

Send that data to your local reportstream and get back a report id:

```
curl -X POST -H "client:simple_report" -H "Content-Type: text/csv"  --data-binary "@./junk/mix.csv" "http://localhost:7071/api/reports"
```

Response:
```
{
  "id" : "7cee139b-bbaa-4b03-abb3-2a4e11ba339c",
 <etc>
}  
```

Local API query to see the Submission details:

`curl "http://localhost:7071/api/waters/report/7cee139b-bbaa-4b03-abb3-2a4e11ba339c/history"
`
If you see reasonable destination data, it works.  If you see a ton of repeats, it failed.  You can even submit the data prior to changing the code to the new branch, then do the history curl to see the bad result, then update the code to this branch, then do the history curl again, to see the good result.

#### How to test in Staging

I have sent a test submission 01a6f9ff-0fd8-465f-9ab5-c38dd34f139a to Staging, that exhibits the problem.    You can try running these steps before and after merging this PR onto Staging.  Before = bad/screwed-up data.   After = hopefully fixed.

Run these three commands:

```
./prime login --env staging
export TOK=$(cat ~/.prime/accesstoken.json | grep token | cut -d"," -f1 | cut -d":" -f2 | tr -d \")
curl -H "authentication-type: okta" -H "authorization: bearer $TOK" "https://staging.prime.cdc.gov/api/waters/report/01a6f9ff-0fd8-465f-9ab5-c38dd34f139a/history"

```
## Changes
- The problem was caused by the fact that a receive Action has multiple reports attached to it.  The query where clause used to constrain to just the report with a non-null sending_org, but that constraint was removed from code; I'm not sure why.  Its now back in.

## Checklist

### Testing
- [N/A] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 

### Process
- [N] Are there licensing issues with any new dependencies introduced?
- [Y] Includes a summary of what a code reviewer should test/verify?
- [N] Updated the release notes?
- [N/A] Database changes are submitted as a separate PR?
- [N/A] DevOps team has been notified if PR requires ops support?

## Fixes 
#6454 

## To Be Done
- I did notice that if the useBatchHeaders is false for a receiver, the bug still surfaces.  CA has useBatchHeaders as false in localhost, so if you generate this data, you'll still see the issue:
./prime data --input-fake 40 --input-schema primedatainput/pdi-covid-19 --output ./junk/mix.csv --target-states 
MD,CA,FL,VT

I believe there are currently no receivers in Prod that use that setting, so this is low priority, I'm hoping.
